handlers/user_handler.go

```diff
@@ -19,8 +17,8 @@ func (h *Handler) UserHandler(w http.ResponseWriter, req *http.Request, _ *model
 	// }
 
 	if err := json.NewEncoder(w).Encode(user); err != nil {
-		logrus.Errorf("error getting user data: %v", err)
-		http.Error(w, "unable to get session", http.StatusInternalServerError)
+		h.log.Error(ErrUserData(err))
```

Comment:
For common errors, like encoding, unmarshalling, it would be more viable to create a generic function to use everywhere, instead of creating individual functions everytime, which are already there in `error.go`.

```
func ErrMarshal(err error) error {
	return utils.ErrMarshal(err)
}
func ErrUnmarshal(err error) error {
	return utils.ErrUnmarshal(err)
}
```

I would suggest few improvements on those functions.
1. Switch from `util` package to a `errors.New(..)` (So it can generate the error while using the Error Code Utility Tool)
2. Include one more string parameter in the function to specify what it was encoding or unmarshalling.

Something like this.
```
func ErrMarshal(err error, obj string) error {
	return errors.New(...)
}
```


This will save a lot of effort in other handlers :wink:

Reply:
Hey @dhruv0000 there are 4 parameters in the error.New so in which parameter should I put this one **Include one more string parameter in the function to specify what it was encoding or unmarshalling.** ?